src/Tmds.ExecFunction/ExecFunction.cs

```diff
@@ -224,7 +226,28 @@ static ExecFunction()
                 }
 
                 HostArguments = PasteArguments.Paste(new string[] { "exec", "--runtimeconfig", runtimeConfigFile, "--depsfile", depsFile, execFunctionAssembly });
-            }
+                if(isTestHost)
+                {
+                    string parentProcessIdRaw = GetApplicationArgument(appArguments, "--parentprocessid");
+                    if(parentProcessIdRaw == null)
+                    {
+                        // this case could be augmented with searching for a dotnet install somewhere
+                        throw new Exception("Can't get dotnet binary location");
```

Comment:
Let's throw `NotSupportedException` instead.

Reply:
Changed to NotSupportedException.